src/Native/Windows/clrcompression/CMakeLists.txt

```diff
@@ -7,6 +7,33 @@ endif()
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 include (GenerateExportHeader)
 
+if($ENV{__BuildArch} STREQUAL x86 OR $ENV{__BuildArch} STREQUAL x64)
+set(NATIVECOMPRESSION_SOURCES
+    clrcompression.def
+    zlib-intel/adler32.c
```

Comment:
e.g. if zlib-intel/adler32.c == zlib/adler32.c then modify these sources to use zlib/adler32.c